src/metabase/task/backfill_query_fields.clj

```diff
@@ -0,0 +1,34 @@
+(ns metabase.task.backfill-query-fields
+  (:require
+   [clojurewerkz.quartzite.jobs :as jobs]
+   [clojurewerkz.quartzite.triggers :as triggers]
+   [metabase.models.query-field :as query-field]
+   [metabase.task :as task]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(defn- backfill-query-fields! []
+  (let [cards (t2/select :model/Card :id [:in {:from      [[:report_card :c]]
+                                                :left-join [[:query_field :f] [:= :f.card_id :c.id]]
+                                                :select    [:c.id]
+                                                :where     [:and
+                                                            [:not :c.archived]
+                                                            [:= :f.id nil]]}])]
+    (run! query-field/update-query-fields-for-card! cards)))
+
+(jobs/defjob ^{org.quartz.DisallowConcurrentExecution true
+               :doc "Backfill QueryField for cards created earlier."}
+  BackfillQueryField [_ctx]
+  (backfill-query-fields!))
+
+(defmethod task/init! ::BackfillQueryField [_]
+  (let [job     (jobs/build
+                  (jobs/of-type BackfillQueryField)
+                  (jobs/with-identity (jobs/key "metabase.task.backfill-query-fields.job"))
+                  (jobs/store-durably))
+        trigger (triggers/build
+                  (triggers/with-identity (triggers/key "metabase.task.backfill-query-fields.trigger"))
+                  ;; runs once on Metabase startup
+                  (triggers/start-now))]
+    (task/schedule-task! job trigger)))
```

Comment:
do we really need to schedule this if it runs "now"?

how is it different from just triggering `backfill-query-fields` in this `task/init!`?

And do you know if this gets repeated run when restarting metabase?

Reply:
Well what we get is sync between metabases not to run it in parallel. Now that I'm thinking about it - it's hard to understand from documentation what `start-now` means. Like `metabase.task.email-remove-legacy-pulse` says "This job will only run once". If so, this seems ideal. :)

Reply:
I tested it, and it indeed ran once.